packages/application-generic/src/services/cache/cache.service.ts

```diff
@@ -95,28 +103,32 @@ export class CacheService implements ICacheService {
 
       pipeline.set(key, value, 'EX', this.getTtlInSeconds(options));
 
-      await this.capturedExec(pipeline, CacheServiceActionsEnum.SET_QUERY, key);
+      return await this.capturedExec(
+        pipeline,
+        CacheServiceActionsEnum.SET_QUERY,
+        key
+      );
     }
   }
 
-  public async keys(pattern?: string) {
+  public async keys(pattern?: string): Promise<string[]> {
```

Comment:
the types are deferred... same above and below

Reply:
I guess you mean inferred.
Personally, I'd rather define the contract for the public API of any class explicit to try to catch the errors as soon as possible. Also it helps me to understand what's going on. Thus I favour verbosity and explicit typing in returns.

Reply:
The downside IMO is that you have to maintain the types anytime anything inside changes. Matt Pocock recommends doing that only when the function has multiple branches or when the return type is complex:
https://www.youtube.com/watch?v=nwSe95uFN8E